src/main/java/org/cryptomator/cryptofs/ch/CleartextFileChannel.java

```diff
@@ -148,17 +151,21 @@ private long writeLockedInternal(ByteSource src, long position) throws IOExcepti
 			assert len <= cleartextChunkSize;
 			if (offsetInChunk == 0 && len == cleartextChunkSize) {
 				// complete chunk, no need to load and decrypt from file
-				ChunkData chunkData = ChunkData.emptyWithSize(cleartextChunkSize);
-				chunkData.copyData().from(src);
-				chunkCache.set(chunkIndex, chunkData);
+				ByteBuffer cleartextChunk = bufferPool.getCleartextBuffer();
```

Comment:
should be named `cleartextBuffer`

Reply:
But it _is_ a chunk. I prefer to name variables according to semantics rather than to types.

Reply:
Correct, but from a technical perspective a chunk is an object derived from the `Chunk` class wrapping a buffer. Here those two definitions intersect, and to avoid confusion I would tend to the technical description. I mean, we are wrapping three lines later a "chunk" with "Chunk". And at creation, the "chunk" is not a "chunk" but just a buffer (the method even says it). Additionally, the naming scheme is also used in other places.

As a compromise, how about renaming it to `cleartextChunkData` ?